### ISSUE TITLE
Listening to multiple topics?

### ISSUE THREAD

**Issue by prullenbak** [salmanzafar949/MQTT_Laravel#18]
Is it possible to listen to multiple topics? 

when I do 

```
$mqtt->ConnectAndSubscribe("something/sometopic", [$this, 'doSomething']);        
$mqtt->ConnectAndSubscribe("something/someothertopic", [$this, 'doSomethingElse']);
```

Only messages on the first topic will call the function. If I switch them, again only the first one will run.

Subscribing to "something/+" and then calling the appropriate function based on figuring out the topic name, works. But it feels wrong.

Any idea? I tried to look for a way to connect once and then calling some subscribe method multiple times, but couldn't find a way to do that. 

Thanks!

**Comment by salmanzafar949** [comment]
yes I've already made changes related to this it's still in progress..
you can test the multiple topic subscription from development branch and let me know.
Testing needs to be done

Thanks

**Comment by prullenbak** [comment]
Thanks for the quick reply!

Unfortunately it doesnt work.

I think the problem now is, that while the method seems to support an array of topics, it still supports only one method? I would like to respond to one topic differently than another. 

(also this leads to an error, but that's another 'topic' _(hehe)_ entirely)

I would love to try to help, but unfortunately I don't have te time at the moment.

**Comment by salmanzafar949** [comment]
same here i also don't have time will work on it once i get time.

**Comment by salmanzafar949** [comment]
fixed this in v2.0.0

**Comment by hb0x01** [comment]
> Is it possible to listen to multiple topics?
> 
> when I do
> 
> ```
> $mqtt->ConnectAndSubscribe("something/sometopic", [$this, 'doSomething']);        
> $mqtt->ConnectAndSubscribe("something/someothertopic", [$this, 'doSomethingElse']);
> ```
> 
> Only messages on the first topic will call the function. If I switch them, again only the first one will run.
> 
> Subscribing to "something/+" and then calling the appropriate function based on figuring out the topic name, works. But it feels wrong.
> 
> Any idea? I tried to look for a way to connect once and then calling some subscribe method multiple times, but couldn't find a way to do that.
> 
> Thanks!

Hi there, can you help me on how to subscribe a topic. I cant understand what am i doing. Can you give me an example? Thanks
